pkg/grpc/universalapi/api_workflow.go

```diff
@@ -0,0 +1,201 @@
+		InstanceID: in.InstanceId,
+	}
+
+	var inputMap map[string]interface{}
+	json.Unmarshal(in.Input, &inputMap)
+	req := workflows.StartRequest{
+		WorkflowReference: wf,
+		Options:           in.Options,
+		WorkflowName:      in.WorkflowName,
+		Input:             inputMap,
+	}
+
+	resp, err := workflowRun.Start(ctx, &req)
+	if err != nil {
+		err = status.Errorf(codes.Internal, fmt.Sprintf(messages.ErrStartWorkflow, err))
+		a.Logger.Debug(err)
+		return &runtimev1pb.WorkflowReference{}, err
+	}
+	ret := &runtimev1pb.WorkflowReference{
+		InstanceId: resp.InstanceID,
+	}
+	return ret, nil
+}
+
+func (a *UniversalAPI) TerminateWorkflowAlpha1(ctx context.Context, in *runtimev1pb.TerminateWorkflowRequest) (*runtimev1pb.TerminateWorkflowResponse, error) {
+	if in.InstanceId == "" {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrMissingOrEmptyInstance)
+		a.Logger.Debug(err)
+		return &runtimev1pb.TerminateWorkflowResponse{}, err
+	}
+
+	if in.WorkflowComponent == "" {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrNoOrMissingWorkflowComponent)
+		a.Logger.Debug(err)
+		return &runtimev1pb.TerminateWorkflowResponse{}, err
+	}
+
+	workflowRun := a.WorkflowComponents[in.WorkflowComponent]
+	if workflowRun == nil {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrWorkflowComponentDoesNotExist, in.WorkflowComponent))
+		a.Logger.Debug(err)
+		return &runtimev1pb.TerminateWorkflowResponse{}, err
+	}
+
+	req := workflows.WorkflowReference{
+		InstanceID: in.InstanceId,
+	}
+
+	err := workflowRun.Terminate(ctx, &req)
+	if err != nil {
+		err = status.Errorf(codes.Internal, fmt.Sprintf(messages.ErrTerminateWorkflow, err))
+		a.Logger.Debug(err)
+		return &runtimev1pb.TerminateWorkflowResponse{}, nil
+	}
+	return &runtimev1pb.TerminateWorkflowResponse{}, nil
+}
+
+func (a *UniversalAPI) RaiseEventWorkflowAlpha1(ctx context.Context, in *runtimev1pb.RaiseEventWorkflowRequest) (*runtimev1pb.RaiseEventWorkflowResponse, error) {
+	if in.InstanceId == "" {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrMissingOrEmptyInstance)
+		a.Logger.Debug(err)
+		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+	}
+
+	if in.EventName == "" {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrMissingWorkflowEventName)
+		a.Logger.Debug(err)
+		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+	}
+
+	if in.WorkflowComponent == "" {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrNoOrMissingWorkflowComponent)
+		a.Logger.Debug(err)
+		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+	}
+
+	workflowRun := a.WorkflowComponents[in.WorkflowComponent]
+	if workflowRun == nil {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrWorkflowComponentDoesNotExist, in.WorkflowComponent))
+		a.Logger.Debug(err)
+		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+	}
+
+	var inputMap map[string]interface{}
+	json.Unmarshal(in.Input, &inputMap)
```

Comment:
Related to my comment on the other PR, the API shouldn't be attempting to interpret the data payload. Rather, it should pass it directly from the API input (as `[]byte`) to the component request (same `[]byte` data, untouched). This will make the API more robust, more performant, and capable of supporting encodings other than JSON if desired in the future.